web/src/components/onboarding/OnboardingTaxId.tsx

```diff
@@ -46,6 +71,7 @@ export const OnboardingTaxId = ({
   const [locationValue, setLocationValue] = useState(state.profileData[fieldName]?.trim().slice(9, 12) ?? "");
   const [taxIdValue, setTaxIdValue] = useState(state.profileData[fieldName]?.trim().slice(0, 9) ?? "");
   const [errorMap, setErrorMap] = useState<ErrorMapType>({ taxId: false, taxIdLocation: false });
+  const [displayTaxId, setDisplayTaxId] = useState<boolean>(getInitialDisplayType());
```

Comment:
this variable seems a little confusing named to me - when I see a boolean controlling "displayX" I'd expect that when it is false, then the thing is not displayed on the page. But that's not the case here. It's always displaying tax ID - this is just controlling whether it's input-view or password-view, right? I think the variable could be renamed and even you could use a specific descriptive type with two descriptive values (like you even did something similar in the useRef above that's not used!) instead of a boolean ("primitive obsession" code smell)

Reply:
I see you updated it - I now doubly recommend this maneuver because seeing a function called "getInitialDisplayType" that returns a true/false value is super baffling to come across. I ask myself, 'the two display _types_ are true and false? what does that mean?'